src/github.com/getlantern/flashlight/ui/service.go

```diff
@@ -25,39 +37,44 @@ var (
 )
 
 func (s *Service) watch() {
-	// Watch for new messages and sent them to the combined output.
-	for b := range s.Out {
+	// Watch for new messages and send them to the combined output.
+	for msg := range s.out {
+		b, err := newEnvelope(s.Type, msg)
+		if err != nil {
+			log.Error(err.Error())
+		}
 		out <- b
 	}
 }
 
-func Register(name string, helloFn func(func([]byte) error) error) (*Service, error) {
+func Register(t string, newMessage func() interface{}, helloFn helloFnType) (*Service, error) {
```

Comment:
There's a new parameter, newMessage.  This is a function that's used to construct a new service-specific message into which we can unmarshal the JSON.